syntax = "proto3";

service MasterNode{
    rpc Store(StoreRequest) returns (StoreReply){}
    rpc Read(ReadRequest) returns (ReadReply){}    
}

message StoreRequest{
    string file_name = 1;  
    bytes file_content = 2;
}

message StoreReply{
    string reply_msg = 1;    
}

message ReadRequest{
    string file_name = 1;
    string timestamp = 2;
    int32 block_size = 3;
}

message ReadReply{
    bytes reply_file = 1;    
}
